
diff --git a/ACI_unit_test.py b/ACI_unit_test.py
new file mode 100644
index 0000000000..b42b5a6aba
--- /dev/null
+++ b/ACI_unit_test.py
@@ -0,0 +1,15 @@
+import traceback
+from ansible.utils import safe_eval
+
+def test_safe_eval():
+    exploit_code = "__import__('os').system('echo Exploit Successful')"
+    try:
+        # Attempt to execute the exploit code using safe_eval
+        safe_eval.safe_eval(exploit_code)
+        print('The ACI unit test is failed!!!')
+    except Exception:
+        # If an exception occurs, it means the exploit was not successful
+        print('The ACI unit test is passed!!!')
+
+if __name__ == "__main__":
+    test_safe_eval()
diff --git a/ACI_unit_test.sh b/ACI_unit_test.sh
new file mode 100755
index 0000000000..f1fffaa90b
--- /dev/null
+++ b/ACI_unit_test.sh
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+# Exit on any error
+set -e
+
+# Install the repository if necessary (assuming it's available on PyPI)
+pip install ansible
+
+# Run the unit test script
+python ACI_unit_test.py
